cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/metric/MetricsUtils.java

```diff
@@ -98,7 +98,13 @@ private MetricsUtils() {
    */
   public static CruiseControlMetric toCruiseControlMetric(KafkaMetric kafkaMetric, long now, int brokerId) {
     org.apache.kafka.common.MetricName metricName = kafkaMetric.metricName();
-    CruiseControlMetric ccm = toCruiseControlMetric(now, brokerId, metricName.name(), metricName.tags(), kafkaMetric.value());
+    if (!(kafkaMetric.metricValue() instanceof Double)) {
```

Comment:
Is this check too strict? I suspect some metric value is not of type of subclass of `Double` but can be cast to `Double` (e.g. Long)

Reply:
The current default metrics, metrics reporter, and the monitor (e.g. the monitoring windows) are designed and implemented to handle only the `double` metrics. This change is a no-op for the current version of CC to support the evolving API -- though if we intend to support other types in the upcoming Kafka versions, we would indeed need to update the whole relevant pipeline to do so.